#33: Updated defaults to kafka 0.9

bbaugher opened the pull request.

bbaugher commented on recipes/default.rb:
```diff
@@ -80,13 +80,15 @@
-    group node["kafka"]["group"]
-    mode 00700
-    recursive true
+if node["kafka"]["server.properties"].has_key? 'log.dirs'
```
Kafka now has 2 log directory properties (log.dirs and log.dir). If log.dirs is not set it uses log.dir which has a default of /tmp/kafka

bbaugher commented on test/integration/default/serverspec/kafka_offset_monitor_spec.rb:
```diff
@@ -42,17 +42,18 @@
-    groupName = "testNewGroup_" + rand(100000).to_s
-      
+    topicName = "testNewTopic-" + rand(100000).to_s
+    groupName = "testNewGroup-" + rand(100000).to_s
```
Kafka 0.9 produces warning about topics with _ or .

bbaugher commented on test/integration/default/serverspec/kafka_spec.rb:
```diff
@@ -118,22 +101,20 @@
 
     # The producer is a command that allows a user to write input to the console as 'messages' to the topic, separated by new line characters
     # In this case we run the command and write the same message several times over 5s in an attempt to ensure the consumer saw the message
-    IO.popen("/opt/kafka/bin/kafka-console-producer.sh --topic #{topic} --broker-list localhost:6667 2> /dev/null", mode='r+') do |io|
```
This was hanging for me. Switched to these new commands which seemed simpler/easier to use

bbaugher commented on test/integration/default/serverspec/kafka_spec.rb:
```diff
@@ -46,28 +46,11 @@
   it { should be_linked_to '/opt/kafka/config' }
 end
 
-['/var/kafka/data1', '/var/kafka/data2', '/var/kafka/data3'].each do |log_dir|
```
Moved these tests into unit tests

mkwhitacre commented:
If we know the offset monitor isn't going to work with the default should we add some warning/logging/error out to this effect so consumers know as well?

noslowerdna commented on attributes/default.rb:
```diff
@@ -5,8 +5,8 @@
-default["kafka"]["scala_version"] = "2.9.2"
-default["kafka"]["version"] = "0.8.1.1"
+default["kafka"]["scala_version"] = "2.11"
+default["kafka"]["version"] = "0.9.0.0"
```
Update README @ https://github.com/cerner/cerner_kafka#how-to-specify-where-to-download-kafka-from-and-which-version-to-install

bbaugher replied:
Done, https://github.com/cerner/cerner_kafka/commit/c0955cc46b0384d0bdf0c3b3f0f17404edf7a51b

noslowerdna commented on attributes/default.rb:
```diff
@@ -31,7 +31,7 @@
 default["kafka"]["logrotate"]["rotate"] = 5
 
-default["kafka"]["server.properties"]["log.dirs"] = "/tmp/kafka-logs"
+default["kafka"]["server.properties"] = {}
```
Update in README

mkwhitacre replied:
+1 any attributes that had meaning but are now meaningless or need to be set by consumers we should call out as part of an upgrade guide.

bbaugher replied:
Done, https://github.com/cerner/cerner_kafka/commit/c0955cc46b0384d0bdf0c3b3f0f17404edf7a51b

noslowerdna commented:
Maybe also bump default offset monitor version to latest (0.2.1)

noslowerdna commented:
In README might link to http://kafka.apache.org/documentation.html#upgrade_9, as it's critical to temporarily set inter.broker.protocol.version=0.8.2.X for a rolling upgrade.

mkwhitacre commented:
Changing the default seems like a major version bump.

bbaugher commented:
We already did a major version bump when we remove a lot of the default attributes (https://github.com/cerner/cerner_kafka/commit/5674e21e519d65c26b566d108f8f446ec1f115a1).

bbaugher commented:
> If we know the offset monitor isn't going to work with the default should we add some warning/logging/error out to this effect so consumers know as well?

Are you referring to the switch to offset storage in kafka? It looks like by default its still zookeeper ([doc](http://kafka.apache.org/documentation.html#consumerconfigs) look for `offsets.storage`).

bbaugher commented:
Addressed all comments with this, https://github.com/cerner/cerner_kafka/commit/c0955cc46b0384d0bdf0c3b3f0f17404edf7a51b

bbaugher commented:
Added upgrade guide to call out what we changed and was non-passive or what they should look out for, https://github.com/cerner/cerner_kafka/commit/f507c8014874a46acb4f1a15cc2daba9244c0d62

mkwhitacre commented:
+1

noslowerdna commented:
+1

bbaugher commented:
Merging with enough +1s
